shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -490,6 +503,59 @@ - (void)dealloc {
   [super dealloc];
 }
 
+#pragma mark - Status Bar touch event related swizzling
+
+constexpr CGFloat kStandardStatusBarHeight = 20.0;
+
+static IMP __original_sendevent = nil;
+
+void _flutter_sendevent(id self, SEL _cmd, UIEvent *event) {
+  // Call the original implementation.
+  ((void(*)(id,SEL,UIEvent*))__original_sendevent)(self, _cmd, event);
+
+  // Intercept touchBegan events.
+  UITouch *touch = event.allTouches.anyObject;
+  if (touch.phase != UITouchPhaseBegan) {
+    return;
+  }
+
+  // Don't handle the touch for double-height status bars. iOS will intercept
+  // and open the associated app.
+  UIApplication *app = [UIApplication sharedApplication];
+  CGRect statusBarFrame = app.statusBarFrame;
+  if (statusBarFrame.size.height != kStandardStatusBarHeight) {
+    return;
+  }
+
+  // Ignore taps outside the status bar. The view already handles them.
+  UIViewController *viewController = app.delegate.window.rootViewController;
```

Comment:
Another edge case: The `FlutterViewController` may not be a part of the view controller hierarchy from the `rootViewController`.